matlab-package/BoardShim.m

```
@@ -28,12 +28,24 @@
             obj.exit_codes(16) = 'ANOTHER_BOARD_IS_CREATED_ERROR';
             obj.exit_codes(17) = 'GENERAL_ERROR';
             obj.exit_codes(18) = 'SYNC_TIMEOUT_ERROR';
-            if isunix
-                obj.libname = 'libBoardController';
+            
+            if ispc
+                obj.libname = 'BoardController';
+                if not(libisloaded('BoardController'))
+                    loadlibrary('lib/BoardController.dll','inc/board_controller.h')
+                end
             elseif ismac
                 obj.libname = 'libBoardController';
+                if not(libisloaded('libBoardController'))
+                    loadlibrary('lib/libBoardController.dylib','inc/board_controller.h')
+                end
+            elseif isunix
+                obj.libname = 'libBoardController';
+                if not(libisloaded('libBoardController'))
+                    loadlibrary('lib/libBoardController.so', 'inc/board_controller.h')
+                end
             else
-                obj.libname = 'BoardController';
+                disp 'OS not supported!'
```

can we raise an exception here?

---

I have change it to error ('OS not supported!')